### PR TITLE
Pin down z-scale normalized to integer (!) failure by separating the tests

### PR DESCRIPTION
**Pull request by h-mayorquin**
I have this tests failing in #1970 which I have seen failing before. It is not failing in my computer so I want to separate it in its own test to hopefully isolate the failure.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
OK, I pined down an example of failure.

```python
from spikeinterface.preprocessing import normalize_by_quantile, scale, center, zscore
from spikeinterface.core import generate_recording

seed = 1
rec = generate_recording(seed=seed, mode="lazy")
rec_int = scale(rec, dtype="int16", gain=100)

zscore_recording = zscore(rec_int, dtype="int16", int_scale=256, mode="mean+std", seed=seed)
traces = zscore_recording.get_traces(segment_index=0)
trace_mean = np.mean(traces, axis=0)
trace_std = np.std(traces, axis=0)
assert np.all(np.abs(trace_mean) < 1)
assert np.all(np.abs(trace_std - 256) < 1)
```

I am  confused about this:

https://github.com/SpikeInterface/spikeinterface/pull/1437

File: src/spikeinterface/preprocessing/normalize_scale.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def __init__(
             means = means[None, :]
             stds = np.std(random_data, axis=0)
             stds = stds[None, :]
-            gain = 1 / stds
+            gain = 1.0 / stds
```
Comment (samuelgarcia) [inline src/spikeinterface/preprocessing/normalize_scale.py:296]: good catch!

**Comment by samuelgarcia** [comment]
`assert np.all(np.abs(trace_std - 256) < 1)` was too strong maybe 2% tolrance error with this
`assert np.all(np.abs(trace_std - 256) < 5)` should be better

**Comment by samuelgarcia** [comment]
OK for me.
@alejoe91  do you want to check ?

**Comment by h-mayorquin** [comment]
> `assert np.all(np.abs(trace_std - 256) < 1)` was too strong maybe 2% tolrance error with this `assert np.all(np.abs(trace_std - 256) < 5)` should be better

In my simulations the std is fine, is the mean estimate that has a lot of variability.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**